### PR TITLE
feat: show better message when zapper fails

### PR DESCRIPTION
**Pull request by patcito**
Show better message when zapper call fails.

zapfailbettermessage

### CODE REVIEW

**Comment by patcito** [comment]
take 2

zapfailstyle

**Review by nymmrx — APPROVED** [review]
lgtm

File: app/containers/Zapper/saga.js [hunk 1]
```diff
@@ -284,8 +291,15 @@ function* zapOut(action) {
     yield call(web3.eth.sendTransaction, zapOutTransaction);
   } catch (error) {
     console.log('Zap Failed', error);
+    let errorMessage = '';
+    if (error && error.message) {
+      errorMessage = error.message;
+    } else {
+      errorMessage =
+        'This means there is not enough liquidity for this token or Zapper may be down. Try with DAI, ETH, USDC or USDT as they offer the more liquidity. ';
```
Comment (x48-crypto) [inline app/containers/Zapper/saga.js:299]: @dudesahn what do you think about this text? Do you think we should use something more generic or does this work?

**Review by x48-crypto — APPROVED** [review]
Looks good to me from a code standpoint.
Deferring to @dudesahn for testing and vetting the error message content

File: app/containers/Zapper/saga.js (hunk 1 above)
Comment (dudesahn) [reply app/containers/Zapper/saga.js:299]: Do we also display error messages for the other errors? Would be good to do for all of them, and style them in the same way @patcito. I'm pretty sure this is the generic/black-box error, but I know zapper has more specific ones as well.

Comment (patcito) [reply app/containers/Zapper/saga.js:299]: They return a generic error that says "low level erc20 transaction failed" most of the time.

**Pull request merged**